torch/distributed/_tensor/dispatch.py

```diff
@@ -149,14 +149,70 @@ def operator_dispatch(
     needs_redistribute = suggested_input_schema is not op_schema
 
     if mesh is not None and mesh.get_coordinate() is None:
-        # if we are on a non-participating device, we simply return
-        # an empty tensor for now.
-        # TODO: what if the op returns a non-tensor value, what if
-        # the op returns a list of tensors, we need to figure out
-        # a consistent way to handle that, and also need to figure
-        # out if we should communicate the result to non-participating
-        # ranks (i.e. a.sum() -> scalar, maybe we should set to 0)
-        local_results = torch.tensor([])
+        # if we are on a non-participating device, we should return
+        # a default value of some type consistent with other
+        # participating devices. This type can be obtained from
+        # the `FunctionSchema` object.
+        def _get_dtype(spec):
+            if spec.tensor_meta is not None:
+                return spec.tensor_meta.dtype
+            else:
+                raise RuntimeError(
+                    f"{spec} has no tensor metadata."
+                )
+
+        def _default_ret_value(arg):
+            # TODO: Replace this dict of primitive date type
+            # default value with a communication
```

Comment:
I think it should not be a TODO item, we should directly implement this feature, otherwise it's not correct. Try to call dist.allgather_object to gather the result on all ranks and return that instead.

If we don't do this, it would be wrong because once it get into a scalar, DTensor could never control it.